lib/router.js

```diff
@@ -161,9 +197,9 @@ class Router {
     _loadModule(modDef, globals) {
         // First, check if we have a copy of this module in the cache, so that we
         // can share it.
-        const cachedModule = this._modules.get(modDef);
-        if (cachedModule && cachedModule._parentGlobals === globals) {
-            return P.resolve(cachedModule);
+        const cached = this._getCachedModule(modDef, globals);
```

Comment:
This is not very intuitive that `cached` return contains a digest and the two functions (getCachedModule and cacheModule) really depend on each other

Reply:
No, it contains the digest of the module definition (which doesn't change) and the actual module, if found.